api/types/database.go

```diff
@@ -85,12 +84,6 @@ type Database interface {
 	GetAD() AD
 	// GetType returns the database authentication type: self-hosted, RDS, Redshift or Cloud SQL.
 	GetType() string
-	// GetIAMPolicy returns AWS IAM policy for the database.
-	GetIAMPolicy() (string, error)
-	// GetIAMAction returns AWS IAM action needed to connect to the database.
-	GetIAMAction() string
-	// GetIAMResources returns AWS IAM resources that provide access to the database.
-	GetIAMResources() []string
```

Comment:
I've decided to refactor these to `lib/srv/db/common/iam`

`api/types` cannot use internal libs (e.g `awslib.PolicyDocument`). This not only limits the return type but also prevents `GetIAMPolicy` from using `GetIAMAction/Resource` internally. `GetIAMPolicy` today uses separate templates today so the same logic is implemented twice.

The `GetIAMPolicy/Action/Resources` model may not apply to Azure/GCP as well. So moved them out of the generic interface in types.Database